Account for other URIs to be turned into strings

kirkkwang opened the pull request:
This commit will expand the cases beyond just Library of Congress URIs. This includes Getty, Geonames, and RightsStatement.org.

Commit:
Account for other URIs to be turned into strings

This commit will expand the cases beyond just Library of Congress URIs.
This includes Getty, Geonames, and RightsStatement.org.

ShanaLMoore commented on app/indexers/uri_to_string_behavior.rb:
```diff
@@ -37,27 +40,86 @@ def convert_uri_to_value(fields, found_mappings: {})
+      end
+
+      [modified_uri, subject_uri, predicate]
+    end
```
```suggestion
# Extracts components needed for RDF querying based on the URI pattern.
# @param uri [String] the URI to process
# @return [Array<String, String, RDF::URI>] processed URI, subject URI, and predicate URI
def extract_rdf_components(uri)
  uri_handlers = {
    'vocab.getty.edu' => -> (uri) {
      modified_uri = uri.gsub('/page/', '/')
      subject_uri = modified_uri.gsub('https://', 'http://')
      [modified_uri, subject_uri, RDF::URI(DEFAULT_LABEL)]
    },
    
    'sws.geonames.org' => -> (uri) {
      modified_uri = "#{uri}/about.rdf"
      subject_uri = "#{uri.gsub('http://', 'https://')}/"
      [modified_uri, subject_uri, RDF::URI("http://www.geonames.org/ontology#name")]
    },
    
    'rightsstatements.org' => -> (uri) {
      modified_uri = "#{uri.chomp('/').gsub('http://', 'https://').gsub('/vocab/', '/data/')}.ttl"
      [modified_uri, uri, RDF::URI(DEFAULT_LABEL)]
    }
  }
  
  # Find the matching handler or use default
  handler_key = uri_handlers.keys.find { |key| uri.include?(key) }
  
  if handler_key
    uri_handlers[handler_key].call(uri)
  else
    [uri, uri, RDF::URI(DEFAULT_LABEL)]
  end
end
```

non blocking but consider refactoring for readability, maintainability and performance. This lambda pattern would make it easier to add new URI handlers if needed, in the future.

kirkkwang replied:
Cool thanks!

kirkkwang committed:
Change to lambda pattern

This commit was suggested in code review and would make it easier to maintain if we need to add more authorities.

Co-authored-by: Shana Moore <shana.lavina.moore@gmail.com>
